src/library_html5.js

```diff
@@ -2598,6 +2598,157 @@ var LibraryJSEvents = {
 
   emscripten_html5_remove_all_event_listeners: function() {
     JSEvents.removeAllEventListeners();
+  },
+
+  emscripten_request_animation_frame: function(cb, userData) {
+    return requestAnimationFrame(function(timeStamp) {
+      Module['dynCall_idi'](cb, timeStamp, userData);
+    });
+  },
+
+  emscripten_cancel_animation_frame: function(id) {
+    cancelAnimationFrame(id);
+  },
+
+  emscripten_request_animation_frame_loop: function(cb, userData) {
+    function tick(timeStamp) {
+      if (Module['dynCall_idi'](cb, timeStamp, userData)) requestAnimationFrame(tick);
```

Comment:
please put the body on another line, with `{ }` on lines before and after (we don't usually put an interesting body that far to the right in a line)